src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -36201,62 +36186,224 @@ public void UnconstrainedTypeParameter_Return_03()
     static U F18<T, U>(T t) where U : class, T => (U)t;
     static U F19<T, U>(T t) where U : struct, T => (U)t;
     static U F20<T, U>(T t) where U : T, new() => (U)t;
+    static U F21<T, U>(T t) => (U)(object)t;
 }";
             var comp = CreateCompilation(new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition }, parseOptions: TestOptions.Regular8);
-            comp.VerifyDiagnostics();
+
+            // PROTOTYPE(NullableReferenceTypes): There should be a warning for F17
+
+            comp.VerifyDiagnostics(
+                // (20,51): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //     static U F18<T, U>(T t) where U : class, T => (U)t;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(U)t").WithLocation(20, 51),
+                // (20,51): warning CS8603: Possible null reference return.
+                //     static U F18<T, U>(T t) where U : class, T => (U)t;
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "(U)t").WithLocation(20, 51),
+                // (23,35): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //     static U F21<T, U>(T t) => (U)(object)t;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(object)t").WithLocation(23, 35)
+            );
+        }
+
+        [Fact]
+        public void UnconstrainedTypeParameter_Uninitialized()
+        {
+            var source =
+@"
+class C
+{
+    static void F1<T>()
+    {
+        T t;
+        t.ToString(); // 1
+    }
+}
+";
+
+            var comp = CreateCompilation(new[] { source, NonNullTypesAttributesDefinition, NonNullTypesTrue }, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (7,9): error CS0165: Use of unassigned local variable 't'
+                //         t.ToString(); // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "t").WithArguments("t").WithLocation(7, 9)
+            );
+        }
+
+        [Fact]
+        public void UnconstrainedTypeParameter_OutVariable()
```

Comment:
>UnconstrainedTypeParameter_OutVariable [](start = 20, length = 38)

Sorry I wasn't clear. I think the interesting case for out var is the call-site.
```
F3(t, out T t2);
// check null-state of t2 (do we reset to nullable, or do we use the state of variable `t`?)
```
We may already have such test.